xlive/Blam/Engine/input/input_xinput.cpp

```diff
@@ -73,33 +72,44 @@ void input_xinput_clear_rumble_state(void)
 
 void input_xinput_update_rumble_state(void)
 {
-	bool suppress_rumble = g_input_feedback_suppress || *input_suppress_global_get();
-	if (!game_in_progress() || game_time_get_paused())
-	{
-		suppress_rumble = true;
-	}
+	bool global_suppress_rumble = false;
 
-	if (controller_profile_get(_controller_index_0)->field_1658 != 1)
+	if (!g_input_feedback_suppress
+		|| *input_suppress_global_get()
+		|| !game_in_progress()
+		|| game_time_get_paused())
 	{
-		suppress_rumble = true;
+		global_suppress_rumble = true;
 	}
 
-	if (controller_button_state_get(_controller_index_0)->plugged_in)
+	for (uint32 device_index = 0; device_index < k_number_of_controllers; ++device_index)
 	{
-		g_xinput_vibration.wLeftMotorSpeed = (suppress_rumble ? 0 : g_vibration_state[_controller_index_0].wLeftMotorSpeed);
```

Comment:
rumble needs to be set to 0 if it's suppressed incase rumbling is already in progress